key.js

```diff
@@ -1,277 +1,446 @@
+    // 1. incrementGameLevel
+    // 2. getGameLevel
+    // 3. startGame
+    // 4. endGame
+    // 5. isGameRunning
+    // 6. isGameOver
+    // 7. getGameLife
+    // 8. decrementGameLife
+    // 9. resetGameState
+    // 10. getGameAudio
+    // 11. printGameState // just for debugging
+    // 12. <create any other method you feel that makes sense>
+    // III. return an object with all the aforementioned utility methods from 1..9.
+
+    // return {
+    //     incrementGameLevel: () => gameState.levelsInfo.currentLevel++
+    // }
+})();
+// IV. Refactor the code below to use the gameModule above for creating the game logic.
+// Button Animation Logic :
+
+const viewModule = (function () {
+    let secondsInterval;
+    function StartTimer() {
+        secondsInterval = setInterval(() => {
+            gameState.seconds -= 1;
+            DOMStrings.timerDisplay.textContent = `${gameState.seconds}`;
+            if (gameState.seconds <= 0) {
+                clearInterval(secondsInterval);
+                gameState.resetGameTimer();
             }
+        }, 1000)
+    }
+    function StopTimer() {
+        clearInterval(secondsInterval);
+    }
+    const DOMStrings = {
+        elementIDs: {
+            leftButton: '#left',
+            rightButton: '#right',
+            upButton: '#up',
+            downButton: '#down'
+        },
+        displaySequence: '.keyDisplay',
+        container: '.container',
+        mainHeading: '.main_heading',
+        defaultLives: '.default_lives',
+        currentLives: '.lives',
+        mainButton: '.main',
+        liveDisplay: '.heart',
+        timerDisplay: '.timer',
+        keyDirections: '.btns',
+        randomKeyArrows: '.arrow-element',
+        keyboardContents: '.contents'
+    }
+    const resetArrowStyles = () => {
+        [document.querySelector(DOMStrings.elementIDs.upButton),
+        document.querySelector(DOMStrings.elementIDs.downButton),
+        document.querySelector(DOMStrings.elementIDs.leftButton),
+        document.querySelector(DOMStrings.elementIDs.rightButtonButton)].forEach(element => {
+            element.style.transform = '';
+            element.style.boxShadow = '';
+        })
+    }
+    function hideDisplayForRetry() {
+        document.querySelectorAll(DOMStrings.randomKeyArrows).forEach(element => {
+            element.classList.add('display-hidden');
+        })
+        document.querySelector(DOMStrings.defaultLives).classList.add('display-hidden');
+        document.querySelector(DOMStrings.displaySequence).classList.add('display-hidden');
+        document.querySelector(DOMStrings.heart).classList.add('display-hidden');
+        document.querySelector(DOMStrings.timerDisplay).classList.add('display-hidden');
+        document.querySelector(DOMStrings.mainHeading).classList.add('display-hidden');
+        document.querySelector(DOMStrings.keyboardContents).classList.add('display-hidden');
```

Comment:
Put this in an array, and apply the logic in a single `forEach` loop as follows:

```suggestion
       [
          DOMStrings.defaultLives,
          DOMStrings.displaySequence,
          DOMStrings.heart, 
          DOMStrings.timerDisplay, 
          DOMStrings.mainHeading, 
          DOMStrings.keyboardContents
        ].forEach(domString => document.querySelector(domString).classList.add('display-hidden'));
```